symbolic-debuginfo/src/dwarf.rs

```diff
@@ -769,6 +773,18 @@ impl<'d, 'a> DwarfUnit<'d, 'a> {
             let function_size = range_buf[range_buf.len() - 1].end - range_buf[0].begin;
             let function_end = function_address + function_size;
 
+            // We have seen duplicate top-level function entries being yielded from the
+            // [`DwarfFunctionIterator`], which combined with recursively walking its inlinees can
+            // blow past symcache limits.
+            // We suspect the reason is that the the same top-level functions might be defined in
+            // different compile units. We suspect this might be caused by link-time deduplication
+            // which merges templated code that is being generated multiple times in each
+            // compilation unit. We make sure to detect this here, so we can avoid creating these
+            // duplicates as early as possible.
+            if !seen_ranges.insert((function_address, function_size)) {
+                continue;
```

Comment:
Now, you need to set `skipped_depth = Some(depth);` before continuing. See other branches where we skip a function.